smdebug/tensorflow/tensor_ref.py

```diff
@@ -96,8 +99,14 @@ def from_variable(cls, variable, export_name=None, mode=None, original_tensor=No
                 # for mirrored variable value this will be the mirrored variable
                 original_tensor = variable
 
+            if is_tf_version_2x() and tf.executing_eagerly() and isinstance(variable, tf.Variable):
+                # In TF 2.X eager mode, TF throws an error if you try to access a tensor's name.
+                # We need to pass it in as a variable, not a tensor, to maintain the name.
+                tf_obj = variable
+            else:
+                tf_obj = variable.value()
```

Comment:
is it true and correct way if not eagerly executing?

Reply:
if not eagerly executing, it is the same as default behavior of TF 1.x.